CONTRIBUTING.md

```diff
@@ -0,0 +1,278 @@
+main reasonable "rejects"/NACKs from anybody who reviewed the code.
+
+Current list of the project maintainers:
+
+- [Andrew Poelstra](https://github.com/apoelstra)
+- [Steven Roose](https://github.com/stevenroose)
+- [Maxim Orlovsky](https://github.com/dr-orlovsky)
+- [Matt Corallo](https://github.com/TheBlueMatt)
+- [Elichai Turkel](https://github.com/elichai)
+- [Sebastian Geisler](https://github.com/sgeisler)
+- [Sanket Kanjalkar](https://github.com/sanket1729)
+
+Anyone may become a prepository maintaner by:
+- Demonstrating long-term commitment to the project development (multiple 
+  accepted PRs and code review of other's PRs over prolonged period of time, 
+  proven by the rank in 
+  [GitHub project contributors list](https://github.com/rust-bitcoin/rust-bitcoin/graphs/contributors))
+- Contributing code and reviews at high quality, passing peer-review of the
+  existing maintainers and other participants.
+- Willing to perform peer review and participate in closing issues on the 
+  regular basis in the future.
+
+On the other hand, those, who had stopped participating in the project 
+development on the ongoing basis may be, after some "grace period", be removed
+from the project maintainers, if there are other candidates who can replace them
+(such that "2 positive reviews from maintainers" requirement will remain 
+achievable).
+
+
+## Coding conventions
+
+### Formatting
+
+We plan to utilize `rustfmt` for keeping the code formatting consistent. 
+However, this will be a gradual process since a thorough peer review is required
+to make sure that no unintended changes are introduced with the change of 
+formatting. Thus, all PRs introducing large blocks of re-formatted code will not
+be reviewed.
+
+The current plan is to phase it in over multiple commits or even multiple PRs,
+which will introduce no other changes that re-formatting, such that each change 
+may be independently re-reproduced by each reviewer. The first commit should add
+attributes to disable fmt for some parts of the code and a second one does the 
+formatting – so only the first one needs review, the rest will be reproducible.
+
+You may check the [discussion on the formatting](https://github.com/rust-bitcoin/rust-bitcoin/issues/172)
+and [how it is planned to coordinate it with crate refactoring](https://github.com/rust-bitcoin/rust-bitcoin/pull/525)
+
+<!--
+Rust-fmt should be used as a coding style recommendations in general, with a
+default coding style. By default, Rustfmt uses a style which conforms to the
+[Rust style guide][style guide] that has been formalized through the [style RFC
+process][fmt rfcs]. It is also required to run `cargo fmt` to make the code
+formatted according to `rustfmt` parameters 
+-->
+
+### Derivation
+
+Derivations applied to a data structures should be standardized:
+
+1. All non-error types should opportunistically derive, where it is possible,
+   the following traits:
+   - `Copy`
```

Comment:
Maybe also mention this is only reasonable for "small" structs (in addition to the no-heap requirement).

Reply:
I'm not really sure if it helps in practice. Moves are `memcpy` anyway.